public/app/plugins/panel/timeseries/plugins/annotations/AnnotationEditorForm.tsx

```diff
@@ -0,0 +1,177 @@
+      if (onSave) {
+        onSave();
+      }
+      return result;
+    });
+
+    const [updateAnnotationState, updateAnnotation] = useAsyncFn(async (event: AnnotationEventUIModel) => {
+      const result = await panelContext.onAnnotationUpdate!(event);
+      if (onSave) {
+        onSave();
+      }
+      return result;
+    });
+
+    const isUpdatingAnnotation = annotation.id !== undefined;
+    const isRegionAnnotation = annotation.time !== annotation.timeEnd;
+    const operation = isUpdatingAnnotation ? updateAnnotation : createAnnotation;
+    const stateIndicator = isUpdatingAnnotation ? updateAnnotationState : createAnnotationState;
+    const ts = isRegionAnnotation
+      ? `${timeFormatter(annotation.time)} - ${timeFormatter(annotation.timeEnd)}`
+      : timeFormatter(annotation.time);
+
+    const onSubmit = ({ tags, description }: AnnotationEditFormDTO) => {
+      operation({
+        id: annotation.id,
+        tags,
+        description,
+        from: Math.round(annotation.time!),
+        to: Math.round(annotation.timeEnd!),
+      });
+    };
+
+    const form = (
+      <div // Annotation editor
+        ref={ref}
+        className={cx(styles.editor, className)}
+        {...otherProps}
+      >
+        <div className={styles.header}>
+          <HorizontalGroup justify={'space-between'} align={'center'}>
+            <div className={styles.title}>Add annotation</div>
+            <div className={styles.ts}>{ts}</div>
+          </HorizontalGroup>
+        </div>
+        <div className={styles.editorForm}>
+          <Form<AnnotationEditFormDTO>
+            onSubmit={onSubmit}
+            defaultValues={{ description: annotation?.text, tags: annotation?.tags || [] }}
+          >
+            {({ register, errors, control }) => {
+              return (
+                <>
+                  <Field label={'Description'} invalid={!!errors.description} error={errors?.description?.message}>
+                    <TextArea
+                      {...register('description', {
+                        required: 'Annotation description is required',
+                      })}
+                    />
+                  </Field>
+                  <Field label={'Tags'}>
+                    <InputControl
+                      control={control}
+                      name="tags"
+                      render={({ field: { ref, onChange, ...field } }) => {
+                        return (
+                          <TagFilter
+                            allowCustomValue
+                            placeholder="Add tags"
+                            onChange={onChange}
+                            tagOptions={getAnnotationTags}
+                            tags={field.value}
+                          />
+                        );
+                      }}
+                    />
+                  </Field>
+                  <HorizontalGroup justify={'flex-end'}>
```

Comment:
nit: why the curly braces here and below (for the string props)? :)

Reply:
This is Webstorm being a bit too eager ;)